decentraland/unity-renderer
# refactor: settings persistence

## What does this PR change?

`Closes #1482 `

Fixes the preset dropdown that was always showing `Custom` at quality settings menu.
Settings data structures were changed from classes to structs so that `equals` compares values instead of references.
A new persistence architecture was implemented to solve the problem of being unable to default a new/non-existent attribute of a user that already had any settings saved. It saves each setting's attribute into a new player prefs key. This architecture also holds retrocompatibility issues by migrating the previous settings saved to the new persistence schema.

## How to test the changes?

1. Go to: https://play.decentraland.org
2. Change your settings (general, quality & audio) to anything you'd like, preferably something different than default settings
3. Close the settings menu
4. Go to: https://play.decentraland.zone/?renderer-branch=refactor/settings-persistence
5. Open the settings menu, you should have the previous settings you changed (general, quality & audio)
6. Change any setting you like
7. Close the settings menu
8. Restart the app
9. You should have the previous settings
10. Go to quality settings then pick any preset from the dropdown (low, medium, high, ultra)
11. Restart the app
12. The preset dropdown should hold whatever preset you chose

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
